### PR TITLE
LibPDF: Decode text strings more (PDFDocEncoding), decode non-text strings less

### PR DESCRIPTION
**Pull request by nico**
Per 1.7 spec 3.8.1, there are multiple logical text string types:
* text strings
* ASCII strings
* byte strings

Text strings can be in UTF-16BE, PDFDocEncoding, or (since PDF 2.0)
UTF-8.

But byte strings shouldn't be converted but treated as binary
data.

This makes us no longer convert strings used for drawing page text.
TABLE 5.6 "Text-showing operators" lists the operands for text-showing
operators as just "string", not "text string" (even though these strings
confusingly are called "text strings" in the body text), so not doing
this there is correct (and matches other viewers).

We also no longer incorrectly convert strings used for cypto data
(such as passwords), if they start with an UTF-16BE or UTF-8 marker.

https://pdfa.org/understanding-utf-8-in-pdf-2-0/ has a good overview of
this.

Text strings without UTF-16BE and UTF-8 markers are now correctly
decoded as PDFDocEncoding.

### CODE REVIEW

**Comment by nico** [comment]
Before:

Screenshot 2023-11-21 at 10 15 30 PM

After:

Screenshot 2023-11-21 at 10 16 41 PM

(The prefix on the page text is still incorrect, but that's a different bug.)

Before:

```sh
% Build/lagom/bin/pdf /Users/thakis/Downloads/0000/0000956.pdf
PDF Version: 1.7
Number of pages: 4
Title: Communiqu? de Presse ? Sagard NewGen et Eurazeo signent un accord d?exclusivit? en vue d?un investissement au capital de I-TRACING
Author: Agnes.Huyghues-Despointes
Creator: Microsoft? Word 2019
Producer: Microsoft? Word 2019
Creation date: D:20210512100058+02'00'
Modification date: D:20210517100458+02'00'
```

After:

```sh
% Build/lagom/bin/pdf /Users/thakis/Downloads/0000/0000956.pdf
PDF Version: 1.7
Number of pages: 4
Title: Communiqué de Presse – Sagard NewGen et Eurazeo signent un accord d’exclusivité en vue d’un investissement au capital de I-TRACING
Author: Agnes.Huyghues-Despointes
Creator: Microsoft® Word 2019
Producer: Microsoft® Word 2019
Creation date: D:20210512100058+02'00'
Modification date: D:20210517100458+02'00'
```

Before:

```sh
% Build/lagom/bin/pdf /Users/thakis/Downloads/0000/0000755.pdf   
PDF Version: 1.5
Number of pages: 3
Title: 2018_05_BescheinigungEink?nfteDatenschutzbestimmung_docx
Creator: PDF reDirect v2
Producer: PixelPlanet PdfEditor (Version 3.0.0.30)
Creation date: D:20180525120309
Modification date: D:20210709084650+02'00'
```

After:

```sh
% Build/lagom/bin/pdf /Users/thakis/Downloads/0000/0000755.pdf        

PDF Version: 1.5
Number of pages: 3
Title: 2018_05_BescheinigungEinkünfteDatenschutzbestimmung_docx
Creator: PDF reDirect v2
Producer: PixelPlanet PdfEditor (Version 3.0.0.30)
Creation date: D:20180525120309
Modification date: D:20210709084650+02'00'
```

File: Userland/Libraries/LibPDF/Document.cpp [hunk 1]
```diff
@@ -74,6 +75,28 @@ PDFErrorOr<Optional<DeprecatedString>> InfoDict::modification_date() const
...
+}
+
+DeprecatedString Document::text_string_to_utf8(DeprecatedString const& text_string)
+{
+    if (text_string.bytes().starts_with(Array<u8, 2> { 0xfe, 0xff })) {
+        // The string is encoded in UTF16-BE
+        return TextCodec::decoder_for("utf-16be"sv)->to_utf8(text_string).release_value_but_fixme_should_propagate_errors().to_deprecated_string();
+    }
+
+    if (text_string.bytes().starts_with(Array<u8, 3> { 239, 187, 191 })) {
```
Comment (ADKaster) [inline Userland/Libraries/LibPDF/Document.cpp:90]: Any particular reason this check uses decimal while the UTF-16BE one uses hex?

**Review by ADKaster — APPROVED** [review]
Sweet! One nit about some moved code

File: Userland/Libraries/LibPDF/Document.cpp (hunk 1 above)
Comment (nico) [reply Userland/Libraries/LibPDF/Document.cpp:90]: Probably not? I can make them both use the same.

**Pull request merged**